sycl/source/handler.cpp

```diff
@@ -1499,14 +1499,39 @@ void handler::depends_on(const std::vector<event> &Events) {
 }
 
 void handler::depends_on(const detail::EventImplPtr &EventImpl) {
+
   if (!EventImpl)
     return;
   if (EventImpl->isDiscarded()) {
     throw sycl::exception(make_error_code(errc::invalid),
                           "Queue operation cannot depend on discarded event.");
   }
+
+  auto EventGraph = EventImpl->getCommandGraph();
+  if (MQueue && EventGraph) {
+    if (EventGraph->getContext() != MQueue->get_context()) {
+      throw sycl::exception(
+          make_error_code(errc::invalid),
+          "Cannot submit to a queue with a dependency from a graph that is "
+          "associated with a different context.");
+      if (EventGraph->getDevice() != MQueue->get_device()) {
+        throw sycl::exception(
+            make_error_code(errc::invalid),
+            "Cannot submit to a queue with a dependency from a graph that is "
+            "associated with a different device.");
+      }
```

Comment:
`begin_recording()` also checks if the queue is in fusion mode, do we not need to replicate that here?

Also could these checks be done in the call to `beginRecording()`?

Reply:
That's fair we can keep the errors separate for clarity.